internal/tasks/relational_loading/relational_loading_helper.go

```diff
@@ -0,0 +1,361 @@
+package relational_loading
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"github.com/bgadrian/fastfaker/faker"
+	"github.com/couchbaselabs/sirius/internal/cb_sdk"
+	"github.com/couchbaselabs/sirius/internal/db"
+	"github.com/couchbaselabs/sirius/internal/docgenerator"
+	"github.com/couchbaselabs/sirius/internal/err_sirius"
+	"github.com/couchbaselabs/sirius/internal/task_result"
+	"github.com/couchbaselabs/sirius/internal/task_state"
+	"github.com/couchbaselabs/sirius/internal/tasks"
+	"github.com/shettyh/threadpool"
+	"log"
+	"reflect"
+	"strings"
+	"time"
+)
+
+var NumOfCollections int
+
+type RelationalOperationConfig struct {
+	TemplateName   string     `json:"template" doc:"true"`
+	Start          int64      `json:"start" doc:"true"`
+	End            int64      `json:"end" doc:"true"`
+	Exceptions     Exceptions `json:"exceptions,omitempty" doc:"true"`
+	FieldsToChange []string   `json:"fieldsToChange" doc:"true"`
+	DocSize        int        `json:"docSize" doc:"true"`
+	DocType        string     `json:"docType,omitempty" doc:"true"`
+	KeySize        int        `json:"keySize,omitempty" doc:"true"`
+	DocsPerKey     int64      `json:"docsPerKey" doc:"true"`
+}
+
+func (o *RelationalOperationConfig) String() string {
+	if o == nil {
+		return "nil config"
+	}
+	b, err := json.Marshal(&o)
+	log.Println(string(b))
+	if err != nil {
+		return ""
+	}
+	return ""
+}
+
+func ConfigureRelationalOperationConfig(o *RelationalOperationConfig) error {
+	if o == nil {
+		return err_sirius.ParsingRelationalOperatingConfig
+	}
+	if o.DocType == "" {
+		o.DocType = docgenerator.JsonDocument
+	}
+	//if o.KeySize == 0 {
+	//	o.KeySize = 16
+	//}
+	if o.KeySize > docgenerator.DefaultKeySize {
+		o.KeySize = docgenerator.DefaultKeySize
+	}
+	if o.DocSize <= 0 {
+		o.DocSize = docgenerator.DefaultDocSize
+	}
+
+	if o.TemplateName == "" {
+		return err_sirius.InvalidTemplateName
+	}
+
+	if o.Start <= 0 && o.End < o.Start {
+		return err_sirius.MalformedOperationRange
+	}
+	o.Start = int64(o.Start / o.DocsPerKey)
+	o.End = int64(o.End / o.DocsPerKey)
+	return nil
+}
+func ConfigTemplateParameters(o *RelationalOperationConfig) error {
+	switch strings.ToLower(o.TemplateName) {
+	case "ecommerce":
+		if o.DocsPerKey == 0 {
+			o.DocsPerKey = 20
+		}
+		NumOfCollections = 6
+	}
+	return nil
+}
+
+// checkBulkWriteOperation is used to check if the Write operation is on main doc or sub doc
+func checkBulkWriteOperation(operation string) bool {
+	switch operation {
+	case tasks.InsertOperation, tasks.UpsertOperation, tasks.BulkUpsertOperation, tasks.BulkInsertOperation:
+		return true
+	default:
+		return false
+	}
+}
+
+// retrieveLastConfig retrieves the RelationalOperationConfig for the offset for a successful Sirius operation.
+func retrieveLastConfig(r *tasks.Request, offset int64) (RelationalOperationConfig, error) {
```

Comment:
rename helper function with relation prefix
```suggestion
func retrieveLastConfigRelational(r *tasks.Request, offset int64) (RelationalOperationConfig, error) {
```